binary-tree/src/lib.rs

```diff
@@ -51,4 +51,30 @@ mod tests {
             "85fc622076752a6fcda2c886c18058d639066a83473d9684704b5a29455ed2ed"
         )
     }
+
+    #[test]
+    fn test_merkleize_multiple_entries() {
+        let mut tree = Blake3Tree::new();
+        let mut keys = vec![
+            [0u8; KEY_LENGTH],
+            [0u8; KEY_LENGTH],
+            [0u8; KEY_LENGTH],
+            [0u8; KEY_LENGTH],
+        ];
+        keys[1][0] = 0x80;
+        keys[2][0] = 0x01;
+        keys[3][0] = 0x81;
+        for i in 0..keys.len() {
+            let mut new_array = [0u8; 32];
+            new_array[0] = i as u8;
+            tree.insert(Key(keys[i]), Value(new_array))
+        }
+        let hash = tree.merkelize();
+        // This will fail, the result is 8c74de28e6bb6b2296cae37cff16266e2dbf533bc204fa4cb0c237761ae8a2c8,
+        // which is what I get in the go code as well.
+        assert_eq!(
+            hex::encode(hash),
+            "e93c209026b8b00d76062638102ece415028bd104e1d892d5399375a323f2218"
```

Comment:
there's no CI so I'm pointing it out here: this value coming from the python code seems to be incorrect.

Reply:
There is CI, just had to approve the run. We can see it fail now.
Let's check first the Python situation and we can update the hash here after.